internal/controller/management_controller.go

```diff
@@ -147,6 +163,54 @@ func (r *ManagementReconciler) Update(ctx context.Context, management *hmc.Manag
 	return ctrl.Result{}, nil
 }
 
+// checkProviderStatus checks the status of a provider associated with a given
+// ProviderTemplate name.  Since there's no way to determine resource Kind from
+// the given template iterate over all possible provider types only returning a
+// not found error if no resources of any provider type are found.
+func (r *ManagementReconciler) checkProviderStatus(ctx context.Context, providerTemplateName string) error {
+	for _, resourceType := range []string{
+		"coreproviders",
+		"infrastructureproviders",
+		"controlplaneproviders",
+		"bootstrapproviders",
+	} {
+		gvr := schema.GroupVersionResource{
+			Group:    "operator.cluster.x-k8s.io",
+			Version:  "v1alpha2",
+			Resource: resourceType,
+		}
+
+		name, kind, conditions, err := utils.GetConditionsFromResource(ctx, r.DynamicClient, gvr,
+			labels.SelectorFromSet(map[string]string{hmc.FluxHelmChartNameKey: providerTemplateName}).String(),
+		)
+		if err != nil {
+			notFoundErr := utils.ResourceNotFoundError{}
+			if errors.As(err, &notFoundErr) {
+				// Check the next resource type.
+				continue
+			}
+
+			return fmt.Errorf("failed to get status for template: %s: %w", providerTemplateName, err)
+		}
+
+		var falseConditionTypes []string
+		for _, condition := range conditions {
+			if condition.Status != v1.ConditionTrue {
+				falseConditionTypes = append(falseConditionTypes, condition.Type)
+			}
+		}
+
+		if len(falseConditionTypes) > 0 {
+			return fmt.Errorf("%s: %s is not yet ready, has false conditions: %s", name, kind, strings.Join(falseConditionTypes, ", "))
+		}
+
+		return nil
+	}
+
+	// We've exhausted all possible provider types and found no items.
+	return fmt.Errorf("failed to find any provider resources affiliated with template: %s", providerTemplateName)
```

Comment:
some providers might not have any affiliated capi operator resources (currently sveltos), so we shouldn't fail here.